Signal/src/view controllers/MessagesViewController.m

```diff
@@ -488,50 +485,130 @@ - (void)setNavigationTitle
+                                                       titleViewWidth,
+                                                       self.navigationBarSubtitleLabel.frame.size.height);
+    
+    self.navigationItem.leftBarButtonItems = @[
+                                               backItem,
+                                               [[UIBarButtonItem alloc] initWithCustomView:self.navigationBarTitleView],
+                                               ];
+    
     if (self.userLeftGroup) {
         self.navigationItem.rightBarButtonItems = @[];
         return;
     }
 
     NSMutableArray<UIBarButtonItem *> *barButtons = [NSMutableArray new];
     if ([self canCall]) {
-        UIBarButtonItem *callButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"btnPhone--white"]
-                                                                       style:UIBarButtonItemStylePlain
-                                                                      target:self
-                                                                      action:@selector(callAction)];
+        // We use UIButtons with [UIBarButtonItem initWithCustomView:...] instead of
+        // UIBarButtonItem in order to ensure that these buttons are spaced tightly.
+        // The contents of the navigation bar are cramped in this view.
+        UIButton *callButton = [UIButton buttonWithType:UIButtonTypeCustom];
+        // The "phone_white_thin" image is slightly assymetric in order
+        // to avoid an excessive margin against the right edge of the screen.
+        [callButton setImage:[UIImage imageNamed:@"phone_white_thin"]
+                    forState:UIControlStateNormal];
         callButton.accessibilityLabel = NSLocalizedString(@"CALL_LABEL", "Accessibilty label for placing call button");
-        callButton.imageInsets = UIEdgeInsetsMake(0, -10, 0, 10);
-        [barButtons addObject:callButton];
+        [callButton addTarget:self
+                       action:@selector(callAction:)
+             forControlEvents:UIControlEventTouchUpInside];
+        [callButton sizeToFit];
+        [barButtons addObject:[[UIBarButtonItem alloc] initWithCustomView:callButton]];
     } else if ([self.thread isGroupThread]) {
-        UIBarButtonItem *manageGroupButton =
-            [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"contact-options-action"]
-                                             style:UIBarButtonItemStylePlain
-                                            target:self
-                                            action:@selector(didTapManageGroupButton:)];
-        // Hack to shrink button image
-        manageGroupButton.imageInsets = UIEdgeInsetsMake(10, 20, 10, 0);
+        UIButton *manageGroupButton = [UIButton buttonWithType:UIButtonTypeCustom];
+        // The "phone_white_thin" image is slightly assymetric in order
```

Comment:
I'm renaming all of these buttons actually.  I'm going to adopt a naming scheme so that related assets are grouped in the alphabetical sort of the asset cache.